StartEycaActivation/handler.ts

```diff
@@ -0,0 +1,270 @@
+import * as df from "durable-functions";
+import { DurableOrchestrationStatus } from "durable-functions/lib/src/classes";
+import { toError } from "fp-ts/lib/Either";
+import { isLeft } from "fp-ts/lib/Either";
+import { identity } from "fp-ts/lib/function";
+import { fromNullable } from "fp-ts/lib/Option";
+import {
+  fromEither,
+  fromLeft,
+  fromPredicate,
+  TaskEither,
+  taskEither,
+  tryCatch
+} from "fp-ts/lib/TaskEither";
+import { ContextMiddleware } from "io-functions-commons/dist/src/utils/middlewares/context_middleware";
+import { RequiredParamMiddleware } from "io-functions-commons/dist/src/utils/middlewares/required_param";
+import {
+  withRequestMiddlewares,
+  wrapRequestHandler
+} from "io-functions-commons/dist/src/utils/request_middleware";
+import {
+  IResponseErrorConflict,
+  IResponseErrorForbiddenNotAuthorized,
+  IResponseErrorInternal,
+  IResponseSuccessAccepted,
+  IResponseSuccessRedirectToResource,
+  ResponseErrorConflict,
+  ResponseErrorForbiddenNotAuthorized,
+  ResponseErrorInternal,
+  ResponseSuccessAccepted,
+  ResponseSuccessRedirectToResource
+} from "italia-ts-commons/lib/responses";
+import { FiscalCode, NonEmptyString } from "italia-ts-commons/lib/strings";
+import { Card } from "../generated/definitions/Card";
+import { StatusEnum as ActivatedStatusEnum } from "../generated/definitions/CardActivated";
+import { StatusEnum as ExpiredStatusEnum } from "../generated/definitions/CardExpired";
+import {
+  CardPending,
+  StatusEnum as PendingStatusEnum
+} from "../generated/definitions/CardPending";
+import { StatusEnum as RevokedStatusEnum } from "../generated/definitions/CardRevoked";
+import { InstanceId } from "../generated/definitions/InstanceId";
+import { UserCgnModel } from "../models/user_cgn";
+import { UserEycaCardModel } from "../models/user_eyca_card";
+import { OrchestratorInput } from "../UpdateCgnOrchestrator";
+import { isEycaEligible } from "../utils/cgn_checks";
+import { makeEycaOrchestratorId } from "../utils/orchestrators";
+import { checkUpdateCardIsRunning } from "../utils/orchestrators";
+
+type ErrorTypes =
+  | IResponseErrorInternal
+  | IResponseErrorForbiddenNotAuthorized
+  | IResponseErrorConflict;
+type ReturnTypes =
+  | IResponseSuccessAccepted
+  | IResponseSuccessRedirectToResource<InstanceId, InstanceId>
+  | ErrorTypes;
+
+type IStartCgnActivationHandler = (
+  context: Context,
+  fiscalCode: FiscalCode
+) => Promise<ReturnTypes>;
+
+const mapOrchestratorStatus = (
+  orchestratorStatus: DurableOrchestrationStatus
+): TaskEither<IResponseSuccessAccepted, IResponseErrorInternal> => {
+  switch (orchestratorStatus.runtimeStatus) {
+    case df.OrchestrationRuntimeStatus.Pending:
+    case df.OrchestrationRuntimeStatus.Running:
+    case df.OrchestrationRuntimeStatus.ContinuedAsNew:
+      return fromLeft(ResponseSuccessAccepted());
+    default:
+      return taskEither.of(
+        ResponseErrorInternal("Cannot recognize the orchestrator status")
+      );
```

Comment:
I don't understand the purpose of this return.

Reply:
Since the activation orchestrator will run until a final status is reached, if its status is not running or pending the card has been already updated (and we check first card status), so  other statuses makes no sense so we will return an Internal Error (it should neve happens). Let's keep in mind that another PR will add a specific mapping for error status that now seems to miss :)